### PR TITLE
Provider 2.x support

### PR DESCRIPTION
**Pull request by prudhvigodithi**
### Description

- Provider 2.x support, coming from https://github.com/opensearch-project/terraform-provider-opensearch/pull/41, https://github.com/opensearch-project/terraform-provider-opensearch/pull/16 (thanks to  @serge-r)

- This should start with provider 2.0 release that can support `2.x` version of OpenSearch.

- The idea of this PR is to streamline the provider OpenSearch `2.x` support.

- Removed configuring the destinations due to this bug https://github.com/opensearch-project/alerting/issues/1078. The new logic should be added to use notifications.  Related issue to add the notification support https://github.com/opensearch-project/terraform-provider-opensearch/issues/30.

-  Updated the repo README to add details about building and testing the provider.

-    Updated the repo README with supported OpenSearch and OpenSearch Dashboards API coverage.

### Issues Resolved
Part of: https://github.com/opensearch-project/terraform-provider-opensearch/issues/39

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by prudhvigodithi** [comment]
Hey @phillbaker and @serge-r can you please take a look and see how I can fix the tests ? 
Thank you

**Comment by prudhvigodithi** [comment]
Hey @phillbaker,  @serge-r I have fixed the tests and logic that should support the 2.x version, Please review once merged this should be good start to move forward with 2.x support.
Thank you

**Comment by prudhvigodithi** [comment]
Updated the example resource docs and the repo README file. Please check.
Thanks

**Comment by prudhvigodithi** [comment]
Can we move forward with this PR?
Adding @peterzhuamazon @phillbaker @jackson-theisen @serge-r @9rnt @atammy-narmi @bbarani 
Thank you

**Comment by andrejvanderzee** [comment]
@phillbaker @jackson-theisen @peterzhuamazon Can you plz take this a step forward and review? Without 2.x support, this provider has not much time left to be relevant.

**Review by peterzhuamazon — APPROVED** [review]

**Comment by peterzhuamazon** [comment]
Need review from @phillbaker @jackson-theisen as well.

Thanks.

File: .github/workflows/test.yml
Comment (prudhvigodithi) [reply .github/workflows/test.yml:100]: We want to test with security plugin enabled as there some CRUD operations for security features like role, role mapping etc, but at the same time for testing we dont need to import all the cluster certs, hence testing with using the flag `plugins.security.ssl.http.enabled=false`.

Comment (prudhvigodithi) [reply .github/workflows/test.yml:100]: It took some time for me to find this :)

**Comment by prudhvigodithi** [comment]
Thanks @peterzhuamazon and @andrejvanderzee.
I have start with the `v2.0.0-beta`, this should be a good starting point to roll out the provider `2.x` support and streamline the repo for upcoming contributions and releases.

File: provider/resource_opensearch_index_template.go
Comment (arianvp) [inline provider/resource_opensearch_index_template.go:60]: This introduces a nasty bug! `opensearch_index_template` is supposed to call `_template` not `_index_template`.   Now this doesn't behave as documented.  The documentation states that `opensearch_composable_index_template` calls `_index_template` and you should use `opensearch_index_template` on `opensearch <2.0` but now both resources use the same endpoint.

**Pull request merged**
